### PR TITLE
Fix issues when downloading FSO builds from GitHub

### PR DESCRIPTION
**Pull request by asarium**
There were multiple issues when trying to download build archives from GitHub:
  1. The URL validator only checked if the HTTP protocol was used but GitHub uses HTTPS.
  2. GitHub does not allow HEAD requests for their resources. I added a fallback which uses GET if the HEAD request fails.

### CODE REVIEW

File: src/com/fsoinstaller/internet/Connector.java [hunk 1]
```diff
@@ -238,34 +238,62 @@ public URLConnection openConnection(URL url) throws IOException
...
-		if (conn instanceof HttpURLConnection)
-		{
-			response = ((HttpURLConnection) conn).getResponseCode();
-			if (response / 100 == 4 || response / 100 == 5)
-				throw new IOException("Server returned HTTP response code " + response + " for URL " + url);
+		// First try using HEAD and if that's not supported use GET
+		try {
+			return getContentLengthImpl(url, true);
+		} catch(IOException e) {
+			logger.debug("Host failed to retreive content length with HEAD, retrying with GET...");
```
Comment (Goober5000) [inline src/com/fsoinstaller/internet/Connector.java]: Could you make the logger log the exception as well as the message?

File: src/com/fsoinstaller/internet/Connector.java [hunk 2]
```diff
@@ -279,6 +307,16 @@ public long getLastModified(URL url) throws IOException
 		
 		return lastModified;
 	}
+
+	public long getLastModified(URL url) throws IOException {
+		// First try using HEAD and if that's not supported use GET
+		try {
+			return getLastModifiedImpl(url, true);
+		} catch(IOException e) {
+			logger.debug("Host failed to retreive last modifies with HEAD, retrying with GET...");
```
Comment (Goober5000) [inline src/com/fsoinstaller/internet/Connector.java]: Here too.

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Looks good, for the most part (except for change in brace style, but I can ignore that).  Just two small changes requested.

**Comment by asarium** [comment]
I changed the log statements to `info` calls and added the exception to the statement.

**Comment by asarium** [comment]
I didn't notice that I used the wrong brace style. I think I fixed that now,

**Review by Goober5000 — APPROVED** [review]

**Comment by Goober5000** [comment]
:+1:

**Pull request merged**